### ISSUE TITLE
[WKT2/Proj6] : support for ocentric and ographic latitude systems

### ISSUE THREAD

**Issue by thareUSGS** [USGS-Astrogeology/GDAL_scripts#38]
issue: The planetary community commonly supports both geocentric (areocentric) and geographic (aerographic) latitude systems. Currently we use different codes to specify the use of ocentric or ographic. This pattern was originally ocentric were even code and ographic were odd codes. For spherical bodies (or triaxial set to a mean spherical radius), there is basically a duplication. 

Is there a method in WKT2 to define ocentric or ographic latitude systems? If spherical, do we need to supply both codes?

**Comment by desruisseaux** [comment]
There is no way as far as I know to define a two-dimensional planetocentric CRS in strictly compliant WKT 2. A constraint come from [7.5.2 - Coordinate system type, dimension and coordinate data type](http://docs.opengeospatial.org/is/18-010r7/18-010r7.html#46) which restrict the number of spherical dimension to 3. So the following is valid:

```
GeodeticCRS[“...”,
  Datum["...”],
  CS[spherical, 3],
     Axis[“Spherical latitude (U)”, north, ...],
     Axis[“Spherical longitude (V)”, east, ...],
     Axis[“Geocentric radius (r)”, up, ...]]
```

If I'm understanding right, for planetocentric needs we would need to be allowed to set the number of dimension to 2, drop the third `Axis` element and said that the _r_ value is implictely such as the point is on the ellipsoid surface. That would be a departure of ISO 19162 (maybe we could said an _extension_ since it does not break any currently valid WKT 2). My proposal would be to implement such extension in some software, experiment it and submit it at OGC if it seems to address the need.

**Comment by rouault** [comment]
OGC Abstract Topic 2 ( https://docs.opengeospatial.org/as/18-005r4/18-005r4.html ) has a precision in "Table 27: Naming constraints for coordinate system axis" regarding the axis names

CS type | When used   in CRS type | Permitted   coordinate system axis names
-- | -- | --
spherical | geodetic | spherical   latitude, spherical longitude, geocentric radius   or   geocentric   latitude, geodetic longitude, geocentric radius

Perhaps the naming variant with "geocentric latitude" would be better ?

What I'm wondering if it would make sense to have a spherical 3D system where the non-angular axis would be an ellipoidal height ? If there is a use case for that ? Or would people would use the distance since the ellipsoid center / geocentric radius ?

**Comment by desruisseaux** [comment]
Actually [the axis name and coordinate system type come together](https://docs.opengeospatial.org/as/18-005r4/18-005r4.html#table_27): if the CS is ellipsoidal, then the axis names should be "geodetic foo". If the CS is spherical, then the axis names should be "spherical foo" or "geocentric foo". However in the astronomical case, I have been told that usage is rather "planetodetic foo" (instead of geodetic) and "planetocentric foo" (instead of geocentric). If we choose to comply with common astronomy practice, that would be another ISO 19111 departure that could be documented in the wiki page proposed in #39.

I don't think that an ellipsoidal heigh with a spherical CS would make sense. But my proposal is different. It is rather to omit the radius completely, like below (strictly speaking, following is invalid in ISO 19111 model since a spherical CS shall always be 3 dimensional):

```
GeodeticCRS[“...”,
  Datum["...”],
  CS[spherical, 2],
     Axis[“Spherical latitude (U)”, north, ...],
     Axis[“Spherical longitude (V)”, east, ...]]
```

A question would be "where in the space is located a point given by only spherical longitude/latitude, without radius?". My proposal is "on the ellipsoid surface", but it does not imply that we associate an ellipsoidal height to the CS. Instead, I propose that the software compute automatically the appropriate radius value.

**Comment by rouault** [comment]
> But my proposal is different.

Yes, that was clear. My question was more a thought about how extensible that could be to close use cases. Like when we have a (terrestrial) ellipsoidal 2D CS for a Geographic 2D CRS, to get the ellipsoidal 3D CS / Geographic 3D CRS, you just add a ```AXIS[“ellipsoidal height (h)”,up,ORDER[3],LENGTHUNIT[“metre”,1.0]]```. But that might be a topic for later.
